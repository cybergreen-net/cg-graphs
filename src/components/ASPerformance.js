import React, { Component } from 'react';
import { connect } from 'react-redux';
import PlotlyGraph from './Plot.js';
import Select from 'react-select';
import Highlighter from 'react-highlight-words';
import 'react-select/dist/react-select.css';
import '../css/temp.css' //this is temp import - needs to be removed for bundle

import { fetchAsDataIfNeeded, AsIsSelected } from '../actions/ASactions';

export class ASPerformance extends Component {
  constructor(props) {
    super(props)
    this.state = {
      plotlyData: [],
      graphOptions: {
        legend: {x:0, y:1},
        height: 300,
        margin: {
          l: 30,r: 30,
          b: 30,t: 0
        },
        font: {
          size: 9,
          color: '#7f7f7f'
        }
      }
    }
  }


  computeState(props=this.props) {
    let lineColors = [
      'rgb(214, 39, 40)', 'rgb(31, 119, 180)', 'rgb(44, 160, 44)',
      'rgb(255, 127, 14)', 'rgb(238, 130, 238)'
    ]
    if(props.view.isFetched) {
      let plotlyData = props.view.selectorConfig.map(config => {
        if (config.as){
          return this.convertToPlotlySeries(
            config.as,
            props.view.country,
            props.view.risk,
            props.data
          )
        }
      }).filter(value => {return value !== undefined})
      plotlyData.forEach((trace, idx) => {
        trace.line = {color: lineColors[idx]}
      })

      return {
        plotlyData: plotlyData
      }
    }
    return {}
  }


  convertToPlotlySeries(asID, countryID, riskID, dataFromCube) {
    var dataTable = dataFromCube[countryID+'/'+riskID+'/'+asID];
    return {
      x: dataTable.map(row => row.date),
      y: dataTable.map(row => row.count),
      name: this.props.asn[asID].title,
      type: 'scatter',
    }
  }


  componentDidMount() {
    this.props.dispatch(fetchAsDataIfNeeded(
      this.props.view.country,
      this.props.view.risk,
      this.props.view.as,
      this.props.viewId
    ))
  };


  componentWillReceiveProps(nextProps) {
    this.setState(this.computeState(nextProps))
  }


  updateValue(idxOfSelector, selectedAS) {
    if(!selectedAS || selectedAS.value === "") {
      this.props.dispatch(AsIsSelected(idxOfSelector, "", this.props.viewId))
    } else {
      this.props.dispatch(AsIsSelected(
        idxOfSelector,
        selectedAS.value,
        this.props.viewId
      ))
      this.props.dispatch(fetchAsDataIfNeeded(
        this.props.view.country,
        this.props.view.risk,
        selectedAS.value,
        this.props.viewId
      ))
    }
  }


  render() {
    return (
      <div className="graph-div">
        <h3>
          {this.props.risks[this.props.view.risk].title.toUpperCase()} &nbsp; | &nbsp;
          {this.props.countries[this.props.view.country].name.toUpperCase()}
        </h3>
        <PlotlyGraph
          data={this.state.plotlyData}
          graphOptions={this.state.graphOptions}
          graphID={this.props.viewId} />
        {this.props.view.selectorConfig.map((selectInfo, idx) => {
          return <ASSelect
                    asn={Object.values(this.props.asn)}
                    disabled={selectInfo.disabled}
                    onChange={this.updateValue.bind(this, idx)}
                    selectedAS={selectInfo.as}
                    key={idx}
                    />
        })}
      </div>
    );
  }
}

export class ASSelect extends Component {
  constructor(props) {
    super(props)
    this.state= {
      inputValue: ''
    }
  }

  optionRenderer(option) {
    if(!option.label){ return }
    return (
      <Highlighter
        searchWords={[this.state.inputValue]}
        textToHighlight={option.label}
      />
    );
  }


  setInputValue(value) {
    this.setState({
      inputValue: value
    })
  }

  render() {
    const selectOptions = this.props.asn.map(asn => {
      return {
        value: asn.number,
        label: asn.title
      }
    })
    selectOptions.unshift({value: '', label: 'Select an ASN'})
    return (
      <div className="Select-div">
        <Select
          name="asn"
          value={this.props.selectedAS || selectOptions[0]}
          options={selectOptions}
          onChange={this.props.onChange}
          onInputChange={this.setInputValue.bind(this)}
          optionRenderer={this.optionRenderer.bind(this)}
          disabled={this.props.disabled}
          clearable={false}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    data: state.entities.cubeByRiskByASN,
    asn: state.entities.asn,
    risks: state.entities.risks,
    countries: state.entities.countries
  }
}

export default connect(mapStateToProps)(ASPerformance)
