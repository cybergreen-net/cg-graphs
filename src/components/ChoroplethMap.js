/* global GLOSSARYPAGE */
import React, {
  Component
} from 'react';
import update from 'react/lib/update'
import {
  connect
} from 'react-redux';
import PlotlyGraph from './Plot.js';
import numeral from 'numeral';
import Select from 'react-select';

import {
  riskAndDateAreSelected,
  fetchData
} from '../actions/ChoroplethMapActions';


export class ChoroplethMap extends Component {
  constructor(props) {
    super(props)
    this.state = {
      graphID: 'ChoroplethMap',
      data: [{
        type: 'choropleth',
        locationmode: 'country names',
        locations: [],
        z: [],
        text: [],
        hoverinfo: 'text',
        autocolorscale: false,
        colorbar: {
          thickness: 5,
          len: 0.5
        },
        colorscale: [
          [0.0, 'rgba(122, 184, 0, 0.7)'],
          ['0.111111111111', 'rgba(147, 189, 12, 0.7)'],
          ['0.222222222222', 'rgba(172, 194, 25, 0.7)'],
          ['0.333333333333', 'rgba(197, 199, 38, 0.7)'],
          ['0.444444444444', 'rgba(223, 204, 51, 0.7)'],
          ['0.555555555556', 'rgba(238, 186, 53, 0.7)'],
          ['0.666666666667', 'rgba(242, 140, 40, 0.7)'],
          ['0.777777777778', 'rgba(246, 95, 27, 0.7)'],
          ['0.888888888889', 'rgba(250, 49, 14, 0.7)'],
          [1.0, 'rgba(250, 0, 0, 0.7)']
        ]
      }],
      layout: {
        height: 600,
        geo: {
          showcountries: true,
          showframe: false,
          showcoastlines: false,
          projection: {
            type: 'robinson'
          }
        },
        margin: {
          t: 0,
          l: 0,
          b: 0,
          r: 0
        }
      }
    }

    this.handleChangeRisk = this.handleChange.bind(this, 'riskToShow')
    this.handleChangeDate = this.handleChange.bind(this, 'dateToShow')
  }


  handleChange(idx, object) {
    if (idx === 'riskToShow') {
      this.props.dispatch(fetchData(
        object.value,
        this.props.view.date
      ))
      this.props.dispatch(riskAndDateAreSelected(
        object.value,
        this.props.view.date
      ))
    } else if (idx === 'dateToShow') {
      this.props.dispatch(fetchData(
        this.props.view.risk,
        object.value
      ))
      this.props.dispatch(riskAndDateAreSelected(
        this.props.view.risk,
        object.value
      ))
    }
  }


  computeState(props = this.props) {
    let locations = []
    let counts = []
    let hoverInfo = []
    let text;
    if (props.view.risk === 100) {
      text = props.view.unit + ' DDOS Potential'
    } else {
      text = 'Count of ' + props.risks[props.view.risk].taxonomy + ' devices'
    }
    if (!props.view.isFetching && props.view.isFetched) {
      props.data[props.view.risk][props.view.date].forEach(entry => {
        if (props.countries[entry.country]) {
          locations.push(props.countries[entry.country].name)
          counts.push(entry[props.view.measure] / props.view.unitDevider)
          hoverInfo.push(numeral(entry[props.view.measure] / props.view.unitDevider).format('0,0') + ' | ' + text + ' in ' + entry.country)
        }
      })

      let max = Math.max(...counts)
      let z = counts.map(item => {
        return (100 * (Math.log(Math.ceil(item)) / Math.log(max)))
      })

      return update(this.state, {
        data: [{
          locations: {
            $set: locations
          },
          z: {
            $set: z
          },
          text: {
            $set: hoverInfo
          }
        }]
      })
    }

    return {}
  }


  componentDidMount() {
    this.props.dispatch(fetchData(this.props.view.risk, this.props.view.date))
  }


  componentWillReceiveProps(nextProps) {
    this.setState(this.computeState(nextProps))
  }


  render() {
    const riskSelectOptions = Object.values(this.props.risks).map(risk => {
      return {
        value: risk.id,
        label: risk.title
      }
    })
    return ( <
      div >
      <
      PlotlyGraph data = {
        this.state.data
      }
      graphOptions = {
        this.state.layout
      }
      graphID = {
        this.state.graphID
      }
      /> <
      div >
      Level of risk posed to others on selected risk on a scale from 0 - 100(100 = worst).For more on data sources, calculations and terms see <
      a href = {
        GLOSSARYPAGE
      } > Glossary and data page < /a> <
      /div> <
      div className = "row" >
      <
      div className = "col-sm-2 col-sm-offset-5"
      title = "Select a risk" >
      <
      Select value = {
        this.props.view.risk
      }
      name = 'riskToShow'
      onChange = {
        this.handleChangeRisk
      }
      clearable = {
        false
      }
      options = {
        riskSelectOptions
      }
      /> <
      /div> <
      /div>

      <
      /div>
    )
  }
}


const mapStateToProps = (state) => {
  return {
    view: state.ChoroplethMapViews,
    data: state.entities.cubeByRiskByDate,
    countries: state.entities.countries,
    risks: state.entities.risks
  }
}


export default connect(mapStateToProps)(ChoroplethMap);
