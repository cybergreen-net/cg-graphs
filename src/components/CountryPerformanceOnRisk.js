import React, { Component } from 'react';
import { connect } from 'react-redux'
import PlotlyGraph from './Plot.js';
import Select from 'react-select';
import 'react-select/dist/react-select.css';
import { countryIsSelected, setViews, fetchDataIfNeeded } from '../actions/cubeActions';


export class CountryPerformanceOnRisk extends Component {
  constructor(props) {
    super(props)
    this.state = {
      cubeByRiskByCountry: {},
      graphOptions: {},
      countries: {},
      selectorConfig: [],
      plotlyData: []
    }
  }


  computeState(props=this.props) {
    let plotlyData = []
    if (props.view.isFetched) {
      plotlyData = props.view.selectorConfig.map(config => {
        if (config.country){
          return this.convertToPlotlySeries(config.country, 1, props.cubeByRiskByCountry)
        }
      }).filter(value => {return value !== undefined})
    }
    let state = {
      cubeByRiskByCountry: props.cubeByRiskByCountry,
      graphOptions: props.graphOptions,
      plotlyData: plotlyData,
      selectorConfig: props.view.selectorConfig
    }
    return state
  }


  convertToPlotlySeries(countryID, riskID, cubeByRiskByCountry) {
    var dataTable = cubeByRiskByCountry[riskID][countryID];
    return {
      x: dataTable.map(row => row.date),
      y: dataTable.map(row => row.count),
      name: this.props.countries[countryID].name,
      type: 'scatter',
    }
  }


  componentDidMount() {
    this.props.dispatch(fetchDataIfNeeded(
      this.props.view.country,
      this.props.view.risk
    ))
    this.props.dispatch(fetchDataIfNeeded(
      't',
      this.props.view.risk
    ))
  };


  componentWillReceiveProps(nextProps) {
    this.setState(this.computeState(nextProps))
  }


  updateValue(idxOfSelector, selectedCountry) {
    if(!selectedCountry || selectedCountry.value === "") {
      this.props.dispatch(countryIsSelected(idxOfSelector, ""))
    } else {
      this.props.dispatch(countryIsSelected(idxOfSelector, selectedCountry.value))
      this.props.dispatch(fetchDataIfNeeded(selectedCountry.value, this.props.view.risk))
    }
  }


  render() {
    let self = this
    return (
      <div>
        <PlotlyGraph
          data={this.state.plotlyData}
          graphOptions={this.state.graphOptions}
          graphID='DDOS-graph' />
        {this.state.selectorConfig.map((selectInfo, idx) => {
          return <CountrySelect
                    countries={Object.values(this.props.countries)}
                    disabled={selectInfo.disabled}
                    onChange={self.updateValue.bind(self, idx)}
                    selectedCountry={selectInfo.country}
                    key={idx}
                    />
        })}
      </div>
    );
  }
}


export class CountrySelect extends Component {
  constructor(props) {
    super(props)
  }


  render() {
    const style = { width: "20%", display: "inline", float: "left" }
    const selectOptions = this.props.countries.map(country => {
      return {
        value: country.id,
        label: country.name
      }
    })
    selectOptions.unshift({value: '', label: 'Select a country'})
    return (
      <div style={style}>
        <Select
          name="countries"
          value={this.props.selectedCountry || selectOptions[0]}
          options={selectOptions}
          onChange={this.props.onChange}
          disabled={this.props.disabled}
        />
      </div>
    );
  }
}


const mapStateToProps = (state, ownProps) => {
  return {
    cubeByRiskByCountry: state.entities.cubeByRiskByCountry,
    countries: state.entities.countries,
    graphOptions: state.entities.layouts,
    view: state.countryPerformanceOnRiskViews[ownProps.view.id]
  }
}


export default connect(mapStateToProps)(CountryPerformanceOnRisk)
