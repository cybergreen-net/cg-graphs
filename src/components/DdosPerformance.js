import React, { Component } from 'react';
import setDataFromUrl from './../utils/Points.js'
import Plot from './Plot.js'

class DdosPerformance extends Component {
  constructor(props) {
    super(props)
    this.state = {
      data: []
    }
  }
  componentDidMount() {
    for (var i = 0; i < this.props.urls.length; i++) {
      setDataFromUrl(this, this.props.urls[i].url, this.props.urls[i].name);
    }
  };

  render() {
    return (
      <div>
        <Plot data={this.state.data} graphOptions={{barmode: 'stack'}} graphID={'ddos'}/>
      </div>
    );
  }
}

export default DdosPerformance;
