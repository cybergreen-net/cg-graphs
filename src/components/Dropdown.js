import React, { Component } from 'react';
import Select from 'react-select';
import 'react-select/dist/react-select.css';
import axios from 'axios'

export default class Dropdown extends Component {
  constructor(props) {
    super(props)
    this.state = { selectedCountry: undefined }
  }

  onChange(country) {
    this.setState({
      selectedCountry: country
    });
    if (country.slug) {
      window.location = `https://cybergreen-staging.herokuapp.com/country/${country.slug}`
    }
  }

  render() {
    const style = { width: "50%", margin: "auto" }
    const selectOptions = Object.values(this.props.countries).map(country => {
      return {
        value: country.id,
        label: country.name,
        slug: country.slug
      }
    })

    selectOptions.unshift({value: '', label: 'Select a country'})
    return (
      <div style={style}>
        <Select
          name="jump to a country page"
          value={this.state.selectedCountry || selectOptions[0]}
          options={selectOptions}
          onChange={this.onChange.bind(this)}
        />
      </div>
    );
  }
}
