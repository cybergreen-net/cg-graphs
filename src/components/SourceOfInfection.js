import React, { Component } from 'react';
import { connect } from 'react-redux'
import PlotlyGraph from './Plot.js';


export class SourceOfInfection extends Component {
  constructor(props) {
    super(props)
    this.state = {
      graphOptions: {
        barmode: 'stack',
        hovermode:'closest',
        showlegend: false,
        height: 252,
        margin: {
          l: 40,r: 30,
          b: 30,t: 0
        },
        xaxis: {
          gridcolor: 'transparent',
          tickformat: '%Y-%m-%d'
        },
        yaxis: {
          title: this.props.view.yLabel
        },
        font: {
          size: 9,
          color: '#7f7f7f'
        }
      },
      plotlyData: []
    }
  }


  computeState(props=this.props) {
    let plotlyData = []

    props.data[props.view.risk][props.view.country].forEach(dataEntry => {
      let colorPallet = [
        'rgb(96, 3, 212)', 'rgb(84, 114, 222)', 'rgb(169, 244, 252)',
        'rgb(7, 101, 240)', 'rgb(0, 212, 154)'
      ]
      let countAllRest = parseInt(dataEntry[props.view.measure])
      dataEntry.as.forEach((asn, idx) => {
        countAllRest -= parseInt(asn[props.view.measure])
        let trace = this.plotlySeries(dataEntry, asn, props.view.measure)
        trace['marker'] = {
          color: colorPallet[idx]
        }
        plotlyData.push(trace)
      })
      let traceAllRest = {
        x: [dataEntry.date],
        y: [countAllRest],
        type: 'bar',
        name: 'Rest',
        marker: {color: 'rgb(200, 2, 16)'},
        text: ['Rest'],
        hoverinfo: 'x+y+text'
      }
      plotlyData.unshift(traceAllRest)
    })

    return {plotlyData: plotlyData}
  }


  plotlySeries(data, asn, measure){
    return {
      x: [data.date],
      y: [asn[measure]],
      type: 'bar',
      name: asn.id,
      text: [asn.id],
      hoverinfo: 'x+y+text'
    }
  }


  componentDidMount() {
    if(this.props.data[this.props.view.risk] && this.props.data[this.props.view.risk][this.props.view.country]) {
      this.setState(this.computeState(this.props))
    }
  };


  componentWillReceiveProps(nextProps) {
    if(nextProps.data[this.props.view.risk] && nextProps.data[this.props.view.risk][this.props.view.country]) {
      this.setState(this.computeState(nextProps))
    }
  }


  render() {
    return (
      <div>
        <h3>
          {this.props.risks[this.props.view.risk].title.toUpperCase()}
          &nbsp; | &nbsp; ASN SOURCE
        </h3>
        <PlotlyGraph
          data={this.state.plotlyData}
          graphOptions={this.state.graphOptions}
          graphID={this.props.viewId}
          clickable={true} />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    data: state.entities.cubeByRiskByCountry,
    risks: state.entities.risks
  }
}

export default connect(mapStateToProps)(SourceOfInfection)
