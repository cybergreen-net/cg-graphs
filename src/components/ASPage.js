import React, { Component } from 'react';
import { connect } from 'react-redux';
import ASPerformance from './ASPerformance';


export class ASPage extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div>
        {Object.values(this.props.views).map(view => {
          return (
            <div key={view.id} className="row">
              <div className="panel panel-default container">
                <ASPerformance view={view} viewId={view.id}/>
              </div>
            </div>
          )
        })}
      </div>
    );
  }
}


const mapStateToProps = (state) => {
  return {
    views: state.ASPerformanceViews
  }
}


export default connect(mapStateToProps)(ASPage)
