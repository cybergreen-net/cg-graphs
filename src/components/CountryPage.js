import React, { Component } from 'react';
import { connect } from 'react-redux';
import CountryPerformanceOnRisk from './CountryPerformanceOnRisk';
import SourceOfInfection from './SourceOfInfection';
import '../css/temp.css' //this is temp import - needs to be removed for bundle

import Highlighter from 'react-highlight-words'
import Select from 'react-select';
import Loader from 'halogen/BounceLoader'
import 'react-select/dist/react-select.css';


export class CountryPage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      selectedASN: undefined,
      inputValue: '',
      spinner: '0px'
    }
  }


  onChange(asn) {
    if(asn.constructor !== Array) {
      this.setState({
        selectedASN: asn
      });
      if (asn !== null && asn.value !== '') {
        this.setState({spinner: '100px'})
        window.location = `asn/${asn.value}`
      }
    }
  }


  optionRenderer(option) {
    if(!option.label){ return }
    return (
      <Highlighter
        searchWords={[this.state.inputValue]}
        textToHighlight={option.label}
      />
    );
  }


  setInputValue(value) {
    this.setState({
      inputValue: value
    })
  }


  download() {
    let country = Object.values(this.props.views)[0].country
    window.location = `/api/v1/count_by_country?limit=500&country=${country}&format=csv`
  }


  render() {
    const selectOptions = Object.values(this.props.asn).map(asn => {
      return {
        value: asn.number,
        label: asn.number + ' | ' + asn.title
      }
    })
    selectOptions.unshift({value: '', label: 'Select an AS'})
    let spinner = Object.values(this.props.views).find(view => {
      return view.isFetching
    })
    return (
      <div>
        { spinner ? <Loader size='100px' color='#00D49A' className='spinner'/> : '' }
        <Loader size={this.state.spinner} color='#00D49A' className='spinner'/>
        <div className="row">
          <div className="col-md-6"></div>
          <div className="col-md-6">
            <Select
              name="asn-selector"
              value={this.state.selectedASN || selectOptions[0]}
              options={selectOptions}
              onChange={this.onChange.bind(this)}
              onInputChange={this.setInputValue.bind(this)}
              optionRenderer={this.optionRenderer.bind(this)}
            />
          </div>
        </div>
        {Object.values(this.props.views).map(view => {
          return (
            <div key={view.id+1} className="row">
              <div className="col-md-6 panel panel-default">
                <CountryPerformanceOnRisk view={view} viewId={view.id}/>
              </div>
              <div className="col-md-6 panel panel-default">
                <SourceOfInfection view={view} viewId={'SOI/'+view.id}/>
              </div>
            </div>
          )
        })}
        <button type='button' className='btn btn-primary-black btn-lg' onClick={this.download.bind(this)}>
          Download
        </button>
        <a href='/download'  style={{float:'right'}}>Get AS data on the Download page</a>
      </div>
    );
  }
}


const mapStateToProps = (state) => {
  return {
    views: state.countryPerformanceOnRiskViews,
    asn: state.entities.asn
  }
}


export default connect(mapStateToProps)(CountryPage)
