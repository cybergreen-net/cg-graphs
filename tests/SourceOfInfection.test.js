import { SourceOfInfection } from '../src/components/SourceOfInfection';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Component is working', () => {
  function setup() {
    let props = {
      risks: {1: {title: 'Open DNS'}},
      view: {
        id: 'GB/1',
        country: 'GB',
        risk: 1,
        AS: {
          id: [192, 254],
          isFetched: false,
          isFetching: false,
          didFailed: false
        }
      },
      data: {
        1: {
          GB: [
            {
              country: "GB",
              risk: 1,
              date: "2017-01-01",
              granularity: "",
              count: 2000,
              count_normalized: 123,
              count_amplified: "321",
              as: [
                {
                  id: 192,
                  count: 500,
                  count_amplified: ""
                },
                {
                  id: 254,
                  count: 550,
                  count_amplified: ""
                },
                {
                  id: 214,
                  count: 780,
                  count_amplified: ""
                }
              ]
            },
            {
              country: "GB",
              risk: 1,
              date: "2017-02-01",
              granularity: "",
              count: 2000,
              count_normalized: 123,
              count_amplified: "321",
              as: [
                {
                  id: 192,
                  count: 440,
                  count_amplified: ""
                },
                {
                  id: 254,
                  count: 620,
                  count_amplified: ""
                },
                {
                  id: 214,
                  count: 710,
                  count_amplified: ""
                }
              ]
            }
          ]
        }
      }
    }

    const enzymeWrapper = shallow(<SourceOfInfection {...props} />)

    return {
      props,
      enzymeWrapper
    }
  }

  it('Renders expected DOM', () => {
    const { enzymeWrapper } = setup()
    expect(toJson(enzymeWrapper)).toMatchSnapshot();
  })

  it('computeState works', () => {
    const { enzymeWrapper, props } = setup()
    let out = enzymeWrapper.instance().computeState(props)
    expect(out.plotlyData.length).toEqual(6)
    expect(out.plotlyData[0].name).toEqual(props.view.AS.id[0])
    expect(out.plotlyData[1].name).toEqual(props.view.AS.id[1])
  })

})
