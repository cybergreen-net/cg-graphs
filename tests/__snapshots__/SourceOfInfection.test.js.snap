exports[`Component is working Renders expected DOM 1`] = `
<div>
  <h3>
    OPEN DNS
      |   ASN SOURCE
  </h3>
  <PlotlyGraph
    clickable={true}
    data={Array []}
    graphOptions={
      Object {
        "barmode": "stack",
        "font": Object {
          "color": "#7f7f7f",
          "size": 9,
        },
        "height": 234,
        "hovermode": "closest",
        "margin": Object {
          "b": 30,
          "l": 30,
          "r": 30,
          "t": 0,
        },
        "showlegend": false,
        "xaxis": Object {
          "gridcolor": "transparent",
          "tickformat": "%Y",
        },
      }
    } />
</div>
`;
